pydicom/pixel_data_handlers/numpy_handler.py

```diff
@@ -1,164 +1,412 @@
+    if len(arr.shape) > 1:
+        raise ValueError("Only 1D arrays are supported.")
 
-    AttributeError
-        if the decoded amount of data does not match the expected amount
+    # The array length must be a multiple of 8, pad the end
+    if arr.shape[0] % 8:
+        arr = np.append(arr, np.zeros(8 - arr.shape[0] % 8))
+
+    # Reshape so each row is 8 bits
+    arr = np.reshape(arr, (-1, 8))
+    if 'PyPy' not in python_implementation():
+        arr = np.fliplr(arr)
+        arr = np.packbits(arr.astype('uint8'))
+        bytestream = arr.tobytes()
+    else:
+        # Implementation for PyPy as it lacks np.packbits
+        def _convert_to_decimal(x):
+            """Return a decimal from the length 8 binary array."""
+            return np.sum(x * [1, 2, 4, 8, 16, 32, 64, 128])
+
+        arr = np.apply_along_axis(_convert_to_decimal, axis=1, arr=arr)
+        bytestream = arr.astype('uint8').tobytes()
+
+    return bytestream
+
+
+def unpack_bits(bytestream):
+    """Unpack bit packed pixel data into a numpy ndarray.
+
+    Suitable for use when (0028,0011) *Bits Allocated* is 1.
+
+    Parameters
+    ----------
+    bytestream : bytes
+        The bit packed pixel data.
 
     Returns
     -------
     numpy.ndarray
-       The contents of the Pixel Data element (7FE0,0010) as an ndarray.
+        The unpacked pixel data as a 1D array.
+
+    Notes
+    -----
+    The implementation for PyPy is roughly 100 times slower than the
+    standard ``numpy.unpackbits`` method.
+
+    References
+    ----------
+    DICOM Standard, Part 5, Section 8.1.1 and Annex D
     """
-    if (dicom_dataset.file_meta.TransferSyntaxUID not in
-            NumpySupportedTransferSyntaxes):
-        raise NotImplementedError("Pixel Data is compressed in a "
-                                  "format pydicom does not yet handle. "
-                                  "Cannot return array. Pydicom might "
-                                  "be able to convert the pixel data "
-                                  "using GDCM if it is installed.")
-    if not have_numpy:
-        msg = ("The Numpy package is required to use pixel_array, and "
-               "numpy could not be imported.")
-        raise ImportError(msg)
-    if 'PixelData' not in dicom_dataset:
-        raise TypeError("No pixel data found in this dataset.")
-
-    # Make NumPy format code, e.g. "uint16", "int32" etc
-    # from two pieces of info:
-    # dicom_dataset.PixelRepresentation -- 0 for unsigned, 1 for signed;
-    # dicom_dataset.BitsAllocated -- 8, 16, or 32
-    if dicom_dataset.BitsAllocated == 1:
-        # single bits are used for representation of binary data
-        format_str = 'uint8'
-    elif dicom_dataset.PixelRepresentation == 0:
-        format_str = 'uint{}'.format(dicom_dataset.BitsAllocated)
-    elif dicom_dataset.PixelRepresentation == 1:
-        format_str = 'int{}'.format(dicom_dataset.BitsAllocated)
+    if 'PyPy' not in python_implementation():
```

Comment:
As far as I remember, PyPy3 implements `unpackbits`, only PyPy2 does not

Reply:
Hopefully fixed, but we don't have a build for PyPy3

Reply:
Ah, and if I correctly understand the [compatibilty page](https://pypy.org/compat.html), both PyPy2 and PyPy3 shall support NumPy, so may be we can adapt the travis script.